hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/management/JpaDistributionSetTagManagement.java

```diff
@@ -95,33 +100,34 @@ public Optional<DistributionSetTag> getByName(final String name) {
             ConcurrencyFailureException.class }, maxAttempts = Constants.TX_RT_MAX, backoff = @Backoff(delay = Constants.TX_RT_DELAY))
     public DistributionSetTag create(final TagCreate c) {
         final JpaTagCreate create = (JpaTagCreate) c;
-        return distributionSetTagRepository.save(create.buildDistributionSetTag());
+        return distributionSetTagRepository.save(AccessController.Operation.CREATE, create.buildDistributionSetTag());
     }
 
     @Override
     @Transactional
     @Retryable(include = {
             ConcurrencyFailureException.class }, maxAttempts = Constants.TX_RT_MAX, backoff = @Backoff(delay = Constants.TX_RT_DELAY))
     public List<DistributionSetTag> create(final Collection<TagCreate> dst) {
-
         @SuppressWarnings({ "rawtypes", "unchecked" })
         final Collection<JpaTagCreate> creates = (Collection) dst;
 
-        return Collections.unmodifiableList(
-                creates.stream().map(create -> distributionSetTagRepository.save(create.buildDistributionSetTag()))
-                        .collect(Collectors.toList()));
+        return creates.stream()
+                .map(create -> distributionSetTagRepository
+                        .save(AccessController.Operation.CREATE, create.buildDistributionSetTag()))
+                .map(DistributionSetTag.class::cast)
+                .toList();
```

Comment:
I would suggest performing a bulk insert here:

    public List<DistributionSetTag> create(final Collection<TagCreate> dst) {
        final List<JpaDistributionSetTag> toCreate = dst.stream().map(JpaTagCreate.class::cast)
                .map(JpaTagCreate::buildDistributionSetTag).toList();
        return Collections
                .unmodifiableList(distributionSetTagRepository.saveAll(AccessController.Operation.CREATE, toCreate));
    }

Reply:
Accepted and will apply the proposal. However, I don't think that bulk import works with hawkBit anyway - since the id generation strategy. Do you think it works?